#
# Submitted-By: Sascha Hauer <s.hauer@pengutronix.de>
# Committed-By: Sascha Hauer <s.hauer@pengutronix.de>
#
# Do not bail out with an error when the chosen node is
# missing. U-Boot expects to have a dtb without chosen
# node, so this is a common usecase and not an error
#
# State:
#
# not submitted 
#

---
 livetree.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: dtc-git-20070301/livetree.c
===================================================================
--- dtc-git-20070301.orig/livetree.c
+++ dtc-git-20070301/livetree.c
@@ -588,8 +588,8 @@ static int check_chosen(struct node *roo
 
 	chosen = get_subnode(root, "chosen");
 	if (! chosen) {
-		ERRMSG("Missing /chosen node\n");
-		return 0;
+		WARNMSG("Missing /chosen node\n");
+		return ok;
 	}
 
 	CHECK_HAVE_WARN_STRING(chosen, "bootargs");
